File: exercises/4thu/isogram.js
function isogram(string) {
  var strArr = string.toLowerCase().split('');
  var letters = {}
  count = 0
  for (var i=0;i<strArr.length;i++) {
    if (!letters[strArr[i]]) {
      letters[strArr[i]] = 1
    } else {
      return false
    }
    count++
  }
  return true
}

var aba = 'abA'
var abc = 'abc'
var dermatoglyphics = 'Dermatoglyphics'
var supercalifragilisticexpialidocious = 'supercalifragilisticexpialidocious'
console.log(aba + ':', isogram(aba), count);
console.log(abc + ':', isogram(abc), count);
console.log(dermatoglyphics + ':', isogram(dermatoglyphics), count);
console.log(supercalifragilisticexpialidocious + ':', isogram(supercalifragilisticexpialidocious), count);

File: exercises/4thu/isogram.rb
def isogram(string)
  string_arr = string.downcase.split('')
  str_hash = {}
  count = 0
  string_arr.each do |let|
    if !str_hash[let]
      str_hash[let] = 1
    else
      return "#{string}: #{false} #{count}"
    end
    count += 1
  end
  return "#{string}: #{true} #{count}"
end


puts isogram('aba')
puts isogram('abc')
